src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeHelpersTests.cs

```diff
@@ -192,16 +191,105 @@ private static void FillStack(int depth)
             }
         }
 
+        public static IEnumerable<object[]> GetUninitializedObject_NegativeTestCases()
+        {
+            // TODO: Test actual function pointer types when typeof(delegate*<...>) support is available
+
+            yield return new[] { typeof(string), typeof(ArgumentException) }; // variable-length type
+            yield return new[] { typeof(int[]), typeof(ArgumentException) }; // variable-length type
+            yield return new[] { typeof(int[,]), typeof(ArgumentException) }; // variable-length type
+            yield return new[] { Array.CreateInstance(typeof(int), new[] { 1 }, new[] { 1 }).GetType(), typeof(ArgumentException) }; // variable-length type (non-szarray)
+            yield return new[] { typeof(Array), typeof(MemberAccessException) }; // abstract type
+            yield return new[] { typeof(Enum), typeof(MemberAccessException) }; // abstract type
+
+            yield return new[] { typeof(Stream), typeof(MemberAccessException) }; // abstract type
+            yield return new[] { typeof(IDisposable), typeof(MemberAccessException) }; // interface type
+
+            yield return new[] { typeof(List<>), typeof(MemberAccessException) }; // open generic type
+            yield return new[] { typeof(List<>).GetGenericArguments()[0], PlatformDetection.IsMonoRuntime ? typeof(MemberAccessException) : typeof(ArgumentException) }; // 'T' placeholder typedesc
```

Comment:
Is this difference something that's worth fixing?

Reply:
Honestly I'd prefer to normalize absolutely everything to ArgumentException, but figured there might be people relying on the old behavior.

Reply:
Turns out `Activator.CreateInstance` and `RuntimeHelpers.GetUninitializedObject` perform many of the same checks, but they throw different exceptions. For example, when given an abstract class, `CreateInstance` throws `MissingMethodException`, but `GetUninitializedObject` throws `MemberAccessException`. Unifying these is going to be a bear.